pkg/subroutines/creator.go

```diff
@@ -0,0 +1,97 @@
+package subroutines
+
+import (
+	"context"
+	"fmt"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/openmfp/golang-commons/controller/lifecycle"
+	"github.com/openmfp/golang-commons/errors"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	v1alpha1 "github.com/openmfp/account-operator/api/v1alpha1"
+)
+
+type CreatorSubroutine struct {
+	client  openfgav1.OpenFGAServiceClient
+	storeID string
+}
+
+func NewCreatorSubroutine(fgaGrpcAddr, storeId string) *CreatorSubroutine {
+	conn, _ := grpc.NewClient(fgaGrpcAddr,
+		grpc.EmptyDialOption{},
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithStatsHandler(otelgrpc.NewClientHandler()),
+	)
+
+	cl := openfgav1.NewOpenFGAServiceClient(conn)
+
+	return &CreatorSubroutine{client: cl, storeID: storeId}
+}
+
+func (e *CreatorSubroutine) Process(ctx context.Context, runtimeObj lifecycle.RuntimeObject) (ctrl.Result, errors.OperatorError) {
+	account := runtimeObj.(*v1alpha1.Account)
```

Comment:
Here I would also add a check if the `account.Spec.Creation` pointer is actually set, to not run into the issue of a possible invalid nil pointer derefence, which would panic the controller and end up in a crash loop.

If the creator field is empty, we do not need to perform any logic

```go
if account.Spec.Creation == nil {
  return ctrl.Result{}, nil
}
```